File: src/endpoint/Bc.Endpoint/CommentTaking.cs
using System.Threading.Tasks;
using Bc.Contracts.Internals.Endpoint.CommentAnswerNotification.Commands;
using Bc.Contracts.Internals.Endpoint.CommentRegistration.Commands;
using Bc.Contracts.Internals.Endpoint.CommentTaking.Commands;
using NServiceBus;

namespace Bc.Endpoint
{
    public class CommentTakingPolicy : IHandleMessages<TakeComment>
    {
        public async Task Handle(TakeComment message, IMessageHandlerContext context)
        {
            await context.Send(new RegisterComment(
                message.CommentId,
                message.UserName,
                message.UserWebsite,
                message.UserComment,
                message.ArticleFileName,
                message.CommentAddedDate)).ConfigureAwait(false);
            
            await context.Send(new RegisterCommentNotification(
                message.CommentId,
                message.UserEmail,
                message.ArticleFileName)).ConfigureAwait(false);
        }
    }
}

File: src/endpoint/Bc.Endpoint/CommentTaking.fs
module Bc.CommentTaking

open System.Threading.Tasks
open Bc.Contracts.Internals.Endpoint.CommentAnswerNotification.Commands
open Bc.Contracts.Internals.Endpoint.CommentRegistration.Commands
open Bc.Contracts.Internals.Endpoint.CommentTaking.Commands
open NServiceBus
open NServiceBus.Logging

type CommentTakingPolicy() =
    interface IHandleMessages<TakeComment> with
        member this.Handle(message, context) =
            async {

                do! context.Send(RegisterComment(
                                    message.CommentId,
                                    message.UserName,
                                    message.UserWebsite,
                                    message.UserComment,
                                    message.ArticleFileName,
                                    message.CommentAddedDate)) |> Async.AwaitTask

                do! context.Send(RegisterCommentNotification(
                                    message.CommentId,
                                    message.UserEmail,
                                    message.ArticleFileName)) |> Async.AwaitTask

            } |> Async.StartAsTask :> Task
